### PR TITLE
Drag and Drop for Mobile.

### PR DESCRIPTION
**Pull request by msb**
A re-factor of the playlist re-ordering code to use the "react-dnd" framework to handle drag and drop. This not only implements a superior drag and drop experience, but also implements drag and drop (although this is sub-optimal).

I'm not sure there's much value in reviewing commit-wise as I started with a reference implementation to get it working and then re-factored into cleaner components.

Closes: #274

### CODE REVIEW

File: frontend/styleguide.config.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ module.exports = {
   // propsParser: require('react-docgen-typescript').withCustomConfig('./tsconfig.json').parse,
 
   exampleMode: 'expand',
+  skipComponentsWithoutExample: true,
```
Comment (rjw57) [inline frontend/styleguide.config.js]: Why was this added?

Comment (msb) [reply frontend/styleguide.config.js]: So that a seperate section isn't included for DraggableContext.js as this is adequately documented in Draggable.js & Draggable.md

Comment (rjw57) [reply frontend/styleguide.config.js]: So, it strikes me that ignoring components without documentation comments would be better.

Or, be explicit and use [ignore](https://github.com/styleguidist/react-styleguidist/blob/master/docs/Cookbook.md#how-to-exclude-some-components-from-style-guide).
